zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+            teams = user_dict["teams"]
+            for team in teams:
+                if team["name"] != team_name:
+                    continue
+                for channel in team["channels"]:
+                    channel_roles = channel["roles"]
+                    channel_name = channel["name"]
+                    if "channel_admin" in channel_roles:
+                        channel_admins_map[channel_name].append(username)
+                    elif "channel_user" in channel_roles:
+                        channel_members_map[channel_name].append(username)
+
+    def get_invite_only(v: str) -> bool:
+        if v == 'O':
+            return False
+        elif v == 'P':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    streams = []
+    initialize_stream_membership_dicts()
+
+    for channel_dict in channel_data_list:
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(channel_dict['name'])
+        stream_name = channel_dict["name"]
+        invite_only = get_invite_only(channel_dict['type'])
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=channel_dict['display_name'] or channel_dict['name'],
+            description=channel_dict["purpose"] or channel_dict['header'],
+            stream_id=stream_id,
+            deactivated=False,
+            invite_only=invite_only,
+        )
+
+        channel_users = set()
+        for username in channel_admins_map[stream_name]:
+            channel_users.add(user_id_mapper.get(username))
+
+        for username in channel_members_map[stream_name]:
+            channel_users.add(user_id_mapper.get(username))
+
+        if channel_users:
+            subscriber_handler.set_info(
+                stream_id=stream_id,
+                users=channel_users,
+            )
+        streams.append(stream)
+
+    return streams
+
+def make_user_messages(zerver_message: List[ZerverFieldsT],
+                       subscriber_map: Dict[int, Set[int]],
+                       is_pm_data: bool,
+                       mention_map: Dict[int, Set[int]]) -> List[ZerverFieldsT]:
+
+    zerver_usermessage = []
+
+    for message in zerver_message:
+        message_id = message['id']
+        recipient_id = message['recipient']
+        sender_id = message['sender']
+        mention_user_ids = mention_map[message_id]
+        subscriber_ids = subscriber_map.get(recipient_id, set())
+        user_ids = subscriber_ids | {sender_id}
+
+        for user_id in user_ids:
+            is_mentioned = user_id in mention_user_ids
+            user_message = build_user_message(
+                user_id=user_id,
+                message_id=message_id,
+                is_private=is_pm_data,
+                is_mentioned=is_mentioned,
+            )
+            zerver_usermessage.append(user_message)
+
+    return zerver_usermessage
```

Comment:
This function looks identical to the HipChat one, can you refactor it to be part of import_util.py?